src/mongocxx/collection.cpp

```diff
@@ -666,6 +667,19 @@ void collection::drop() {
     }
 }
 
+void collection::read_concern(class read_concern rc) {
+    libmongoc::collection_set_read_concern(_impl->collection_t, rc._impl->read_concern_t);
+}
+
+stdx::optional<class read_concern> collection::read_concern() const {
```

Comment:
So, the spec says: "Unless specified [on a client, database, or collection], the [read_concern] value MUST be inherited from its parent."  We are not currently following this behavior, nor is the C driver.   Because we aren't storing any read concern objects, but are instead calling into the C driver each time we fetch the read_concern, I feel that we shouldn't implement this logic.  We should instead expect the C driver to return the right thing when we call libmongoc::collection_get_read_concern--the "right thing" being a parent's read_concern, if it's not set on the collection.

So I guess I'm not asking you to do anything differently here, I just wanted to bring this up.  I've opened [CXX-758](https://jira.mongodb.org/browse/CXX-758) to track this, there's a corresponding CDRIVER ticket linked.